docs/classes/collection.md

```diff
@@ -1,38 +1,396 @@
+Operations within the Collection class (e.g. [find()](#find),
+[insertOne()](#insertone)) will generally inherit the Collection's options. One
+notable exception to this rule is that [aggregate()](#aggregate) (when not using
+a cursor), [distinct()](#distinct), and the [findAndModify][findandmodify]
+helpers do not yet support a "typeMap" option due to a driver limitation. This
+means that they will always return BSON documents and arrays as stdClass objects
+and arrays, respectively.
 
-Operations within the Collection class (e.g. `find()`, `insertOne()`) will
-generally inherit the Collection's options. One notable exception to this rule
-is that `aggregate()` (when not using a cursor) and the `findAndModify` variants
-do not yet support a type map option due to a driver limitation. This means that
-they will return BSON documents and arrays as `stdClass` objects and PHP arrays,
-respectively.
+[findandmodify]: http://docs.mongodb.org/manual/reference/command/findAndModify/
 
-## Collection-level Operations
+---
 
-The Collection class has methods for collection-level operations, such as
-dropping the collection, CRUD operations, or managing the collection's indexes.
+## __construct()
 
-### Dropping the Collection
+```php
+function __construct(MongoDB\Driver\Manager $manager, $databaseName, $collectionName, array $options = [])
+```
+
+If the Collection is constructed explicitly, any omitted options will be
+inherited from the Manager object. If the Collection is selected from a
+[Client](client.md) or [Database](database.md) object, options will be
+inherited from that object.
+
+### Supported Options
+
+readConcern (MongoDB\Driver\ReadConcern)
+:   The default read concern to use for collection operations. Defaults to the
+    Manager's read concern.
+
+readPreference (MongoDB\Driver\ReadPreference)
+:   The default read preference to use for collection operations. Defaults to
+    the Manager's read preference.
+
+typeMap (array)
+:   Default type map for cursors and BSON documents.
+
+writeConcern (MongoDB\Driver\WriteConcern)
+:   The default write concern to use for collection operations. Defaults to the
+    Manager's write concern.
+
+### See Also
+
+ * [MongoDB\Collection::withOptions()](#withoptions)
+ * [MongoDB\Database::selectCollection()](database.md#selectcollection)
+
+---
+
+## aggregate()
+
+```php
+function aggregate(array $pipeline, array $options = []): Traversable
+```
+
+Executes an aggregation framework pipeline on the collection.
+
+This method's return value depends on the MongoDB server version and the
+"useCursor" option. If "useCursor" is true, a MongoDB\Driver\Cursor will be
+returned; otherwise, an ArrayIterator is returned, which wraps the "result"
+array from the command response document.
+
+**Note:** BSON deserialization of inline aggregation results (i.e. not using a
+command cursor) does not yet support a custom type map. Support is pending new
+functionality in the driver.
```

Comment:
Does this include documents with `_pclass` attributes? If so that might be worth mentioning. The sentence could be construed to mean only that the `type` option is not supported.